api/users/users-router.js

```diff
@@ -1,44 +1,81 @@
 const express = require('express');
 
 // You will need `users-model.js` and `posts-model.js` both
+const Users = require(`../users/users-model`)
+const Posts = require(`../posts/posts-model`)
+
 // The middleware functions also need to be required
+const {logger, validateUserId, validateUser, validatePost} = require(`../middleware/middleware`)
 
 const router = express.Router();
 
-router.get('/', (req, res) => {
-  // RETURN AN ARRAY WITH ALL THE USERS
+router.get('/', (req, res, next) => {
+  Users.get()
+  .then(user => {
+    res.status(200).json(user)
+  })
+  .catch(next)
 });
 
-router.get('/:id', (req, res) => {
+router.get('/:id', validateUserId, (req, res, next) => {
   // RETURN THE USER OBJECT
   // this needs a middleware to verify user id
+  res.json(req.user)
 });
 
-router.post('/', (req, res) => {
+router.post('/', validateUser, (req, res, next) => {
   // RETURN THE NEWLY CREATED USER OBJECT
   // this needs a middleware to check that the request body is valid
+  Users.insert(req.body)
+    .then(user => {
+      res.status(201).json(user)
+    })
+    .catch(next)
```

Comment:
I know they taught you the universal error with this next in the catch. Personally I think it's better to have a specific error message for the request so you can more easily debug where things went wrong. So for my part I don't use next in the router request code since it's the end of the line and I'm able to write a more specific error message which can help me find where the issue is in the code.